### PR TITLE
Agents

### PR DESCRIPTION
**Pull request by ideadapt**
hi
hab mal noch eine komponente gemacht, die die agent status anzeigt. weisst du wie der status für einen offline agent lautet? 'offline'?

mozaik

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
-var request = require('superagent');
+var request = require('request-promise');
```
Comment (swissmanu) [inline src/client.js]: any special intention by switching to `request-promise`?

File: src/client.js [hunk 2]
```diff
@@ -12,40 +10,90 @@ var client = function (mozaik) {
...
+                .then((responses)=>{
                     return {
                         results: responses.map(function(response) {
-                            return response.body.results.result[0];
+                            return response.results.result[0];
                         })
                         , baseUrl: config.get('bamboo.baseUrl')
                     };
-                });
+                }).catch(console.error);
```
Comment (swissmanu) [inline src/client.js]: please use mozaiks logger with `chalk` instead of plain `console.error` (see line 26)

**Comment by swissmanu** [comment]
cool! thanks for the contribution :)
guess they call that state just `offline`... but no sure... would have to take one offline to say it with confidence ;)

--edit--
would you mind to enhance the readme too?

File: src/client.js (hunk 1 above)
Comment (ideadapt) [inline src/client.js]: I think superagent had problems with plain text/html responses.

**Comment by ideadapt** [comment]
I updated the readme :)
Ok, so we have to try. Now the state is expected to be 'offline'.

**Comment by ideadapt** [comment]
I cleaned up components folder by creating components/agents and components/planResults sub folders. Offline detection is now properly implemented and tested :)

**Comment by swissmanu** [comment]
thanks for your contribution @ideadapt :+1:

**Pull request merged**